### ISSUE TITLE
Consider supporting multiple sources for a Resource

### ISSUE THREAD

**Issue by bgilbert** [coreos/ignition#1104]
Hmm that said we should probably also have nicer support for multiple URLs in included Ignition, so that one could upload configs to multiple S3 regions and have fallback; something like:
```
'{"ignition":{"config":{"replace":{"source0":"https://s3-us-east-1/mybucket/myconfig.ign", "source1": "https://s3-us-west-1/mybucket/myconfig.ign", "verification": "sha512here"}},"version":"3.1.0"}}'
```
(Multiple urls should probably require `Verification` to ensure they are the same and avoid nondeterminism)

_Originally posted by @cgwalters in https://github.com/coreos/ignition/pull/1100#issuecomment-700243609_

Edit: per https://github.com/coreos/ignition/issues/1104#issuecomment-701700340, that would be:

```
'{"ignition":{"config":{"replace":{"sources":["https://s3-us-east-1/mybucket/myconfig.ign", "https://s3-us-west-1/mybucket/myconfig.ign"], "verification": "sha512here"}},"version":"3.1.0"}}'
```

**Comment by bgilbert** [comment]
If we wanted to do this, I'm thinking we'd have a `sources` array in Resource.  If we fetch every source in parallel, with the first successful result winning (as we do for config sources in e.g. OpenStack), we can avoid being responsible for a timeout and backoff scheme.

**Comment by paulnivin** [comment]
> If we fetch every source in parallel, with the first successful result winning (as we do for config sources in e.g. OpenStack), we can avoid being responsible for a timeout and backoff scheme.

I think a timeout, backoff, and retry scheme would still be needed for intermittent network issues. For example, a sufficient part of the stack is available to retrieve the initial Ignition config (e.g. from user-data), but access to the referenced sources is not (yet) available (e.g. DNS failure, route failure, etc).

**Comment by bgilbert** [comment]
Right, I didn't phrase that well.  Ignition already has a retry scheme for exactly the reason you mentioned.  But I'd like to avoid adding logic for giving up on one resource and falling back to the next one.

**Comment by bgilbert** [comment]
Closing in favor of #1276.  More info in https://github.com/coreos/ignition/pull/1176#issuecomment-1119113385.
